Realm/RLMValue.mm

```diff
@@ -0,0 +1,134 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2020 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#import "RLMValue.h"
+#import "RLMUtil.hpp"
+
+#pragma mark NSData
+
+@implementation NSData (RLMValue)
+
+- (RLMPropertyType)rlm_valueType {
```

Comment:
This is named oddly

Reply:
as we are adding a property name to the global namespace we cannot risk conflicting with what Apple may add in the future. Adding a 3 character prefix helps us guard against that.